File: android/app/src/main/java/com/lumoralabs/macro/data/SyncManager.kt
package com.lumoralabs.macro.data

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkCapabilities

object SyncManager {
    fun isOnline(context: Context): Boolean {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val network = cm.activeNetwork ?: return false
        val capabilities = cm.getNetworkCapabilities(network) ?: return false
        return capabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
    }

    fun syncGroups(context: Context) {
        if (isOnline(context)) {
            val localGroups = LocalGroupStorage.loadGroups(context)
            localGroups.forEach { SupabaseService.saveGroup(it) }
            SupabaseService.getGroups { cloudGroups ->
                LocalGroupStorage.saveGroups(context, cloudGroups)
            }
        }
    }
}

File: ios/Data/SyncManager.swift
import Foundation
import Combine
import Network

@MainActor
class SyncManager: ObservableObject {
    private let monitor = NWPathMonitor()
    private let queue = DispatchQueue(label: "NetworkMonitor")
    
    @Published var isOnline = false
    @Published var lastSyncDate: Date?

    init() {
        startNetworkMonitoring()
    }
    
    private func startNetworkMonitoring() {
        monitor.pathUpdateHandler = { [weak self] path in
            DispatchQueue.main.async {
                self?.isOnline = path.status == .satisfied
            }
        }
        monitor.start(queue: queue)
    }

    func checkNetworkStatus() async -> Bool {
        await withCheckedContinuation { continuation in
            monitor.pathUpdateHandler = { path in
                continuation.resume(returning: path.status == .satisfied)
            }
        }
    }

    // TODO: Re-enable when types are properly imported
    /*
    func syncGroups(localStorage: LocalGroupStorage, supabaseService: SupabaseService) async {
        guard isOnline else {
            print("🌐 Cannot sync groups - device is offline")
            return
        }
        
        do {
            // Upload local groups to cloud
            let localGroups = localStorage.loadGroups()
            for group in localGroups {
                try await supabaseService.saveGroup(group)
            }
            
            // Download cloud groups and update local storage
            let cloudGroups = try await supabaseService.getGroups()
            localStorage.saveGroups(cloudGroups)
            
            lastSyncDate = Date()
            print("✅ Groups synced successfully")
            
        } catch {
            print("❌ Sync failed: \(error.localizedDescription)")
        }
    }
    */
}

